src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormattingPassBase.cs

```diff
@@ -129,7 +128,12 @@ protected async Task<List<TextChange>> AdjustIndentationAsync(FormattingContext
                 continue;
             }
 
-            var scopeOwner = syntaxTreeRoot.LocateOwner(new SourceChange(originalLocation, 0, string.Empty));
+            if (originalLocation > syntaxTreeRoot.EndPosition)
```

Comment:
@davidwengier I can't say I understand what the original code was doing: the source mappings have an OriginalLocation past the end of the document. `LocateOwner` helpfully returns `null` for this case, so this condition replicates the behavior. However, it doesn't really make sense to me why the source mapping is beyond the end of the document. Any thoughts on that, or should we just proceed with this as is?

Reply:
Ours is not to question why, unless we feel like re-writing the formatting engine :)

Seems that an `originalLocation` that is one character past the end of the document is entirely possible and expected here, so I think this change is fine. It probably just makes other code easier, as it means there will always be a result when the code elsewhere does a binary search of the `sourceMappingMap`.

Reply:
Alright. If that's expected, then this is ready for review.